libraries/aws/shadow/src/aws_iot_shadow_parser.c

```diff
@@ -144,52 +141,53 @@ AwsIotShadowError_t _AwsIotShadow_ParseErrorDocument( const char * pErrorDocumen
                                    ERROR_DOCUMENT_CODE_KEY,
                                    ERROR_DOCUMENT_CODE_KEY_LENGTH,
                                    &pCode,
-                                   &codeLength ) == false )
-    {
-        /* Error parsing JSON document, or no "code" key was found. */
-        IotLogWarn( "Failed to parse code from error document.\n%.*s",
-                    errorDocumentLength,
-                    pErrorDocument );
-
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_BAD_RESPONSE );
-    }
-
-    /* Code must be in error document. */
-    AwsIotShadow_Assert( ( pCode > pErrorDocument ) &&
-                         ( pCode + codeLength < pErrorDocument + errorDocumentLength ) );
-
-    /* Convert the code to an unsigned integer value. */
-    code = ( uint32_t ) strtoul( pCode, NULL, 10 );
-
-    /* Parse the error message and print it. An error document must always contain
-     * a message. */
-    if( AwsIotDocParser_FindValue( pErrorDocument,
-                                   errorDocumentLength,
-                                   ERROR_DOCUMENT_MESSAGE_KEY,
-                                   ERROR_DOCUMENT_MESSAGE_KEY_LENGTH,
-                                   &pMessage,
-                                   &messageLength ) == true )
+                                   &codeLength ) == true )
     {
-        IotLogWarn( "Code %lu: %.*s.",
-                    code,
-                    messageLength,
-                    pMessage );
+        /* Code must be in error document. */
+        AwsIotShadow_Assert( ( pCode > pErrorDocument ) &&
```

Comment:
(Minor) Not sure if this assert check is really necessary. The first expression checks if the code is present after the starting address of the error document. Do you know why that is a requirement? If this function is carrying the knowledge of the JSON encoding format, and is thus, checking that the binary error code's location exists after the starting location, I'm not a fan of this parser to carry serialization format level information.

(I understand that this is out of the scope of the PR, but I would like it to be removed if not required)

Reply:
I agree with you on this. @gordonwang0 Do you think so?

Reply:
@leegeth Can you comment on this?